go-controller/pkg/ovn/controller/apbroute/external_controller.go

```diff
@@ -22,39 +22,64 @@ func (g gatewayInfoList) String() string {
 
 	s := strings.Builder{}
 	for _, item := range g {
-		s.WriteString(fmt.Sprintf("%s, ", item.gws))
+		s.WriteString(fmt.Sprintf("%s, ", item.Gateways))
 	}
 	return s.String()
 }
 
+// GetBFDForIP returns the BFD value for the given IP stored in the gatewayInfoList
+// An IP can only have one BFD value.
+func (g gatewayInfoList) GetBFDForIP(ip string) (bool, error) {
+	for _, i := range g {
+		if i.Gateways.Has(ip) {
+			return i.BFDEnabled, nil
+		}
+	}
+	return false, fmt.Errorf("IP %s not found", ip)
+}
+
+// HasIP returns true if the given IP is found in one of the gatewayInfo elements stored in the gatewayInfoList
 func (g gatewayInfoList) HasIP(ip string) bool {
 	for _, i := range g {
-		if i.gws.Has(ip) {
+		if i.Gateways.Has(ip) {
 			return true
 		}
 	}
 	return false
 }
 
-func (g gatewayInfoList) Insert(items ...*gatewayInfo) (gatewayInfoList, sets.Set[string]) {
+// Insert adds a slice of gatewayInfo's to the gatewayInfolist and ensures that no IP duplicates are inserted. It returns a new
+// gatewayInfoList containing the added gatewayInfo elements and set containing the duplicated IPs found during the Insert operation.
+func (g gatewayInfoList) Insert(items ...*gatewayInfo) (gatewayInfoList, sets.Set[string], error) {
 	ret := append(gatewayInfoList{}, g...)
 	duplicated := sets.New[string]()
 	for _, item := range items {
-		for _, ip := range item.gws.UnsortedList() {
+		gws := sets.Set[string]{}
+		for _, ip := range item.Gateways.UnsortedList() {
 			if ret.HasIP(ip) {
+				v, err := ret.GetBFDForIP(ip)
```

Comment:
nit: you could have a `HasBFDEnabled(ip string) (bool, bool)` that returns both if BFD is enabled or disabled and if the IP is there, much like  it would be if you had a map with that information: `bfdEnabled, hasIP := mapOfIPToBFDEnabled[ip]` 
You would avoid iterating twice through the gatewayInfos here

Reply:
done